lib/config/decrypt.spec.ts

```diff
@@ -53,4 +57,190 @@ describe('config/decrypt', () => {
+          ${'{"o":"fgh",           "r":"",     "v":"zv-2"}'}     | ${'fgh/rp1'} | ${'zv-2'}
+          ${'{"o":"az123/fgh",     "r":"rp1",  "v":"zv-3"}'}     | ${'fgh/rp1'} | ${'zv-3'}
+          ${'{"o":"az123/fgh",     "r":"",     "v":"zv-4"}'}     | ${'fgh/rp1'} | ${'zv-4'}
+          ${'{"o":"az123/*",       "r":"",     "v":"zv-5"}'}     | ${'fgh/rp1'} | ${'zv-5'}
+          ${'{"o":"az123/",        "r":"",     "v":"zv-6"}'}     | ${'fgh/rp1'} | ${null}
+          ${'{"o":"az123",         "r":"",     "v":"zv-7"}'}     | ${'fgh/rp1'} | ${null}
+          ${'{"o":"az1",           "r":"",     "v":"zv-8"}'}     | ${'fgh/rp1'} | ${null}
+          ${'{"o":"az123/any",     "r":"rp1",  "v":"zv-9"}'}     | ${'fgh/rp1'} | ${null}
+          ${'{"o":"az123/any",     "r":"",     "v":"zv-10"}'}    | ${'fgh/rp1'} | ${null}
+          ${'{"o":"any/*",         "r":"",     "v":"zv-11"}'}    | ${'fgh/rp1'} | ${null}
+          ${'{"o":"az123/*,any/*", "r":"",     "v":"zv-12"}'}    | ${'fgh/rp1'} | ${'zv-12'}
+          ${'{"o":"fgh,any/*",     "r":"",     "v":"zv-13"}'}    | ${'fgh/rp1'} | ${'zv-13'}
+          ${'{"o":"az123/,any/*",  "r":"",     "v":"zv-14"}'}    | ${'fgh/rp1'} | ${null}
+          ${'{"o":"any/*,fgh/",    "r":"",     "v":"zv-15"}'}    | ${'fgh/rp1'} | ${'zv-15'}
+          ${'{"o":"any/*,az123",   "r":"",     "v":"zv-16"}'}    | ${'fgh/rp1'} | ${null}
+          ${'{"o":"any/*,az12",    "r":"",     "v":"zv-17"}'}    | ${'fgh/rp1'} | ${null}
+          ${'{"o":"az12,any/*",    "r":"",     "v":"zv-18"}'}    | ${'fgh/rp1'} | ${null}
+        `(
+          'equals("$str", "$repo") === $expected',
+          ({ str, repo, expected }) => {
+            GlobalConfig.set({
+              platform: 'azure',
+              endpoint: 'https://dev.azure.com/az123',
+            });
+            expect(validateDecryptedValue(str, repo)).toBe(expected);
+          },
+        );
+      });
+
+      it('endpoint URL invalid', () => {
+        GlobalConfig.set({
+          platform: 'azure',
+          endpoint: 'ht tps://dev.az ure.com/az123',
+        });
+        expect(
+          validateDecryptedValue(
+            '{"o":"proj", "r":"repo", "v":"any-1"}',
+            'proj/repo',
+          ),
+        ).toBe('any-1');
+        expect(
+          validateDecryptedValue(
+            '{"o":"proj", "r":"", "v":"any-2"}',
+            'proj/repo',
+          ),
+        ).toBe('any-2');
+
+        expect(
+          validateDecryptedValue(
+            '{"o":"col/proj", "r":"", "v":"any"}',
+            'proj/repo',
+          ),
+        ).toBeNull();
+        expect(
+          validateDecryptedValue(
+            '{"o":"col/*", "r":"", "v":"any"}',
+            'proj/repo',
+          ),
+        ).toBeNull();
+      });
+
+      it('endpoint URL without collection', () => {
+        GlobalConfig.set({
+          platform: 'azure',
+          endpoint: 'https://dev.azure.com/',
```

Comment:
test with custom domain?  eg `https://domain.test/some/devops-server/` and  `https://devops.domain.test/`

Reply:
I tried to lookup details on azureDevOps self-host and couldn't find much details/examples.. 

`https://domain.test/some/devops-server/` will result in `pathname: some/devops-server` which will be added as top-level org/collection name and will affect the matching. 
In case `https://domain.test/some/devops-server/` this is a valid server name then I have to take into account and hardcod a check for `https://dev.azure.com/`

Reply:
added handling and tests